### DISCUSSION TITLE
simulation server pool

### DISCUSSION THREAD

**Discussion by mvanriet** [nunobrum/PyLTSpice#72] [Ideas]
For big simulations, or for lot simulating a lot of parameters, it would be interesting to be able to distribute these across several computers.

A server can be started on several servers.
Then simulations can be started with jobs being distributed across these computers automatically.
Netlist and results are automatically copied between client and server.
For the user, it should be the no more complex then running it on your own machine.

For the server : PyLTSpice would contain a server module that can be started using "python -m PyLTSpice --server" or something.
A  ClientSimCommander - with same API as the normal SimCommander - would send the .net-file to the server instead of running it locally.
Server puts it in a directory, run it using SimCommander, zip the results (raw-file and log-file), and send it back.
Client unzips the results, so that it will just look like it was executed locally.
Communication between client/server can be done using a standard RPC protocol, e.g. using JSONRPClib.

**Comment by nunobrum** [comment]
This is an interesting feature and it's implementation doesn't require much change in the present code.
 The actually implementation kind of implements a kind of client server approach, as all simulation are already being executed in a different thread implemented by the class RunTask.
What would be needed, is to write a new class, let's say for example: RunOnServerTask, where the zipped netlist is sent to a server.
In order to select if the RunTask is used, or the RunOnServerTask is used, a useServer() method can be used to add a pool of servers. 
A smart implementation would select from the pool of servers the one with more available resources. To be discussed how this can be implemented.

Additionally the already existing setLTspiceRunCommand() could be used in case there are more than one Spice tool. By the way, this function will be renamed to setSpiceTool()
Also, with this, it should not be assumed that the spice tool is installed locally.
